### PR TITLE
Make the default theme empty so users can extend it

### PR DESCRIPTION
**Pull request by mohsen1**
Similar to styled-components we're allowing users to declare their themes using interface augmentation.

### CODE REVIEW

File: types/react-with-styles/lib/WithStylesContext.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
 import { Context } from "react";
 
-declare const context: Context<{ stylesInterface: any, stylesTheme: any }>;
-export default context;
\ No newline at end of file
+declare const context: Context<{ stylesInterface: any, stylesTheme: any, direction?: any }>;
```
Comment (lencioni) [inline types/react-with-styles/lib/WithStylesContext.d.ts:3]: This seems unrelated

File: types/react-with-styles/index.d.ts [hunk 2]
```diff
@@ -4,13 +4,14 @@
...
 //                 Joe Lencioni <https://github.com/lencioni>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.8
+// TypeScript Version: 3.2
 
 import * as React from 'react';
 import * as PropTypes from 'prop-types';
 import { CSSProperties } from 'aphrodite';
 
-interface Theme { [key: string]: any; }
```
Comment (lencioni) [inline types/react-with-styles/index.d.ts:13]: I'm not sure I understand what this enables. Can you give me an example of what you couldn't do before and after this change? Also, will this cause errors for anyone and make it difficult to update?

Comment (mohsen1) [reply types/react-with-styles/index.d.ts:13]: Removing the index signature makes it possible to extend this interface with type-safe themes. There is an example in the test. With this index signature any other property from theme such as `unitt` would also work.

File: types/react-with-styles/lib/WithStylesContext.d.ts (hunk 1 above)
Comment (mohsen1) [reply types/react-with-styles/lib/WithStylesContext.d.ts:3]: The context provider default value can get a `direction` property too. This is to allow that.

File: types/react-with-styles/index.d.ts (hunk 2 above)
Comment (mohsen1) [reply types/react-with-styles/index.d.ts:13]: with regarding to upgrading:

if users didn't have a type-safe theme definition they can add the index signature themselves:

```ts
declare module 'react-with-styles' {
  interface Theme {
    [key: string]: any;
  }
}
```

File: types/react-with-styles/lib/WithStylesContext.d.ts (hunk 1 above)
Comment (lencioni) [reply types/react-with-styles/lib/WithStylesContext.d.ts:3]: That seems unrelated to "Make the default theme empty so users can extend it"

File: types/react-with-styles/index.d.ts (hunk 2 above)
Comment (lencioni) [reply types/react-with-styles/index.d.ts:13]: Ah I see now. Thank you for explaining.

**Review by lencioni — APPROVED** [review]

**Comment by elibarzilay** [comment]
@mohsen1, are the two comments resolved and this thing mergable?

**Comment by mohsen1** [comment]
@elibarzilay yes. this is good to merge
